### PR TITLE
Fix xl issues

### PR DESCRIPTION
**Pull request by crtrott**
This fixes a few issues in builds using IBM XL. It splits an test object file into two files, and it moves another object file to a different executable. All the failing tests did fail when run with gtest filter, i.e. failure didn't depend on other tests being run first. The explicitly added cpp files are there, since the UVM and HostPinned test cpp files are NOT autogenerated in our system yet.

### CODE REVIEW

File: core/unit_test/Makefile [hunk 1]
```diff
@@ -265,6 +268,7 @@ ifeq ($(KOKKOS_INTERNAL_USE_HIP), 1)
 	OBJ_HIP += TestHIP_MDRange_a.o TestHIP_MDRange_b.o TestHIP_MDRange_c.o TestHIP_MDRange_d.o TestHIP_MDRange_e.o
 	OBJ_HIP += TestHIP_Spaces.o
 	OBJ_HIP += TestHIPHostPinned_ViewAPI_a.o TestHIPHostPinned_ViewAPI_b.o TestHIPHostPinned_ViewAPI_c.o TestHIPHostPinned_ViewAPI_d.o TestHIPHostPinned_ViewAPI_e.o
+        OBJ_HIP += TestHIPHostPinned_ViewCopy_a.o TestHIPHostPinned_ViewCopy_b.o
```
Comment (dalg24) [inline core/unit_test/Makefile:271]: Fix indentation here and below

File: core/unit_test/TestViewCopy_b.hpp [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+bool run_check(ViewType v, typename ViewType::value_type value) {
+  using exec_space = typename ViewType::memory_space::execution_space;
+  int errors       = 0;
+  Kokkos::fence();
+  Kokkos::parallel_reduce(Kokkos::RangePolicy<exec_space>(0, v.extent(0)),
+                          CheckResult<ViewType>(v, value), errors);
+  return errors == 0;
+}
+
+}  // namespace
```
Comment (dalg24) [inline core/unit_test/TestViewCopy_b.hpp:79]: This code is being duplicated in `TestViewCopy_{a,b}.hpp`

File: core/unit_test/CMakeLists.txt [hunk 3]
```diff
@@ -409,6 +417,8 @@ if(Kokkos_ENABLE_HIP)
       hip/TestHIPHostPinned_ViewAPI_c.cpp
       hip/TestHIPHostPinned_ViewAPI_d.cpp
       hip/TestHIPHostPinned_ViewAPI_e.cpp
+      hip/TestHIPHostPinned_ViewCopy_a.cpp
+      hip/TestHIPHostPinned_ViewCopy_b.cpp
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt:421]: Too bad we are not generating these files yet

**Review by dalg24 — CHANGES_REQUESTED** [review]
Fix the indentation in the Makefile

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

**Review by dalg24 — APPROVED** [review]

**Pull request merged**